### DISCUSSION TITLE
Seeking Advice: Running Appium Tests on iOS Without Device Registration

### DISCUSSION THREAD

**Discussion by leedh2004** [appium/appium#20491] [Q&A]
Hello, I need some advice regarding an Appium-based test I'm working on.

There's a limit of 100 device registrations on iOS, and I'm concerned that I might hit this limit soon.

Is there a way to run Appium integration tests without having to register each actual device with the Apple Developer Account?

To bypass this, would it work if I tested the app signed with an Enterprise certificate?

Any guidance on how to proceed would be greatly appreciated! 🙏

**Comment by mykola-mokhnach** [comment]
> There's a limit of 100 device registrations on iOS, and I'm concerned that I might hit this limit soon.

This limitation is set by Apple and only Apple can change it.

> To bypass this, would it work if I tested the app signed with an Enterprise certificate?

in order for such app to start on a device the certificate must be already accepted. A possible workaround for it might be to install **another** app signed with the **same** certificate on the device, accept/trust it and keep the app installed **forever** (uninstalling it would cause iOS to untrust the certificate). Since iOS 18 Apple made it even more complicated, because they require the device restart in order to trust a certificate.

**Comment by nanoscopic** [comment]
The limitation here is not getting the app installed onto more devices. For Appium to work for automation the way it does it uses xcuitest / xctest. That requires the device to be setup as a developer device, and there is a strict 1 year limit of 100 on these. You can only reset the devices once a year.

Generally people work around this by using different developer accounts.

Simulators can be used, but they aren't real devices so there are tradeoffs.
